src/Resolvers/Microsoft.DotNet.NativeWrapper/EnvironmentProvider.cs

```diff
@@ -60,5 +59,35 @@ public string GetCommandPath(string commandName)
 
             return commandPath;
         }
+
+        public string GetDotnetExeDirectory()
+        {
+            string environmentOverride = _getEnvironmentVariable("DOTNET_MSBUILD_SDK_RESOLVER_CLI_DIR");
+            if (!string.IsNullOrEmpty(environmentOverride))
+            {
+                return environmentOverride;
+            }
+
+            var dotnetExe = GetCommandPath("dotnet");
+
+            if (dotnetExe != null && !Interop.RunningOnWindows)
+            {
+                // e.g. on Linux the 'dotnet' command from PATH is a symlink so we need to
+                // resolve it to get the actual path to the binary
+                dotnetExe = Interop.Unix.realpath(dotnetExe) ?? dotnetExe;
+            }
+
+            return Path.GetDirectoryName(dotnetExe);
+        }
+
+        public static string GetDotnetExeDirectory(Func<string, string> _getEnvironmentVariable = null)
```

Comment:
The parameter should be called without the underscore `getEnvironmentVariable` - as is it's VERY confusing (since it has the same name as a member field).